CHANGELOG.md

```diff
@@ -1,5 +1,96 @@
+* Update gitignore
+  ([#4333](https://github.com/microsoft/ApplicationInsights-Java/pull/4333))
+* Change enforcedPlatform to platform
+  ([#4334](https://github.com/microsoft/ApplicationInsights-Java/pull/4334))
+* Remove unused dependencies
+  ([#4335](https://github.com/microsoft/ApplicationInsights-Java/pull/4335))
+* Simplify running against azure-monitor-opentelemetry-autoconfigure snapshots
+  ([#4336](https://github.com/microsoft/ApplicationInsights-Java/pull/4336))
+* Improve snapshot testing
+  ([#4339](https://github.com/microsoft/ApplicationInsights-Java/pull/4339))
+* Add spotlessApply to copilot instructions
+  ([#4341](https://github.com/microsoft/ApplicationInsights-Java/pull/4341))
+* Add devcontainer configuration to reduce Copilot agent build time from 5+ minutes to under 1 minute
+  ([#4342](https://github.com/microsoft/ApplicationInsights-Java/pull/4342))
+* Fix devcontainer
+  ([#4344](https://github.com/microsoft/ApplicationInsights-Java/pull/4344))
+* Better devcontainer
+  ([#4349](https://github.com/microsoft/ApplicationInsights-Java/pull/4349))
+* update gitignore
+  ([#4361](https://github.com/microsoft/ApplicationInsights-Java/pull/4361))
+* Clean up README.md
+  ([#4360](https://github.com/microsoft/ApplicationInsights-Java/pull/4360))
+* Simplify and use codespaces built-in prebuild feature
+  ([#4358](https://github.com/microsoft/ApplicationInsights-Java/pull/4358))
+* Remove unused code
+  ([#4357](https://github.com/microsoft/ApplicationInsights-Java/pull/4357))
+* Sync ErrorProne configuration with OpenTelemetry Java instrumentation
+  ([#4364](https://github.com/microsoft/ApplicationInsights-Java/pull/4364))
+* Sync checkstyle configuration with OpenTelemetry Java Instrumentation
+  ([#4365](https://github.com/microsoft/ApplicationInsights-Java/pull/4365))
+* Sync spotless configuration with OpenTelemetry Java Instrumentation
+  ([#4366](https://github.com/microsoft/ApplicationInsights-Java/pull/4366))
+* Fix warning about toolchains
+  ([#4373](https://github.com/microsoft/ApplicationInsights-Java/pull/4373))
+* Fix grpc smoke test deprecated class usage warnings
+  ([#4372](https://github.com/microsoft/ApplicationInsights-Java/pull/4372))
+* Fix a few gradle warnings
+  ([#4375](https://github.com/microsoft/ApplicationInsights-Java/pull/4375))
+* Suppress warnings for some safe operations
+  ([#4371](https://github.com/microsoft/ApplicationInsights-Java/pull/4371))
+* Update Gradle to latest version 8.14.3
+  ([#4345](https://github.com/microsoft/ApplicationInsights-Java/pull/4345))
+* Update to OpenTelemetry Java Instrumentation 2.18.1
+  ([#4378](https://github.com/microsoft/ApplicationInsights-Java/pull/4378))
+* Fix a couple more gradle deprecations
+  ([#4367](https://github.com/microsoft/ApplicationInsights-Java/pull/4367))
+* Fix deprecated EnvironmentResourceProvider reference
+  ([#4377](https://github.com/microsoft/ApplicationInsights-Java/pull/4377))
```

Comment:
I think this is the only user facing PR here (it solves a deprecation warning that was being printed that users saw)

The other one that we are getting fixed in azure-monitor-opentelemetry-autoconfigure that's worth mentioning since reported by user is: https://github.com/microsoft/ApplicationInsights-Java/pull/4314